### PR TITLE
Refactor BehaviorService to use List<T> instead of ArrayList

### PR DESCRIPTION
**Pull request by elachlan**
Related: #8140

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8187)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 1]
```diff
@@ -597,17 +590,12 @@ private MenuCommand FindCommand(CommandID commandID, IMenuCommandService menuSer
...
+                return _behaviorStack[0];
             }
 
-            if (g != null && g.Behavior != null)
-            {
-                return g.Behavior;
-            }
-
-            return null;
+            return g is not null && g.Behavior is not null ? g.Behavior : null;
```
Comment (halgab) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:598]:
```suggestion
            return g?.Behavior;
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ public sealed partial class BehaviorService : IDisposable
     {
         private readonly IServiceProvider _serviceProvider;             // standard service provider
         private readonly AdornerWindow _adornerWindow;                  // the transparent window all glyphs are drawn to
-        private readonly ArrayList _behaviorStack;                      // the stack behavior objects can be pushed to and popped from
+        private readonly List<Behavior> _behaviorStack;                      // the stack behavior objects can be pushed to and popped from
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:35]: Please align the comments

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs (hunk 1 above)
Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs]: Thanks!

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 3]
```diff
@@ -151,14 +151,7 @@ public Behavior CurrentBehavior
...
             {
-                if (_behaviorStack != null && _behaviorStack.Count > 0)
-                {
-                    return (_behaviorStack[0] as Behavior);
-                }
-                else
-                {
-                    return null;
-                }
+                return _behaviorStack != null && _behaviorStack.Count > 0 ? _behaviorStack[0] : null;
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs:154]:
```suggestion
                return _behaviorStack is not null && _behaviorStack.Count > 0 ? _behaviorStack[0] : null;
```

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**